### PR TITLE
Added a file characters_unique

### PR DESCRIPTION
**Pull request by nishithshowri006**
### **Describe your change:**

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by Kush1101 — COMMENTED** [review]
Please run your code through `Black` code formatted as discussed in [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).

File: strings/charactes_unique.py [hunk 1]
```diff
@@ -18,23 +27,31 @@ def isUniqueSet(s):
         hashSet.add(i)
     return True
 
-def isUniqueHash(s):
+
+def isUniqueHash(s: str) -> str:
```
Comment (Kush1101) [inline strings/charactes_unique.py:31]: Avoid using a single character as a variable names for variables whose scope extends just a few lines. Try making it more descriptive. `string`

File: strings/charactes_unique.py [hunk 2]
```diff
@@ -18,23 +27,31 @@ def isUniqueSet(s):
...
     """
-    This uses Hashes
+    This function uses ascii values as index to determine
+    the string contains unique characters
     s -> string
```
Comment (Kush1101) [inline strings/charactes_unique.py:35]: After having Type hints, there is no need for this.

File: strings/charactes_unique.py (hunk 1 above)
Comment (cclauss) [reply strings/charactes_unique.py]: Strings is the name of the directory,string is the name of a module in the Python standard library, and str is the name of a Python builtin data type so in this case, s is an acceptable.

File: strings/charactes_unique.py (hunk 2 above)
Comment (cclauss) [reply strings/charactes_unique.py]: Sorry but Python 2 died on 1/1/2020.  One of the ways that Python 3 is different than Python 2 is that all str are Unicode.  Please include doctests that demonstrate that this code proper deals with sentences that include ãèíôü😀🎄

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]

File: strings/characters_unique.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    This function uses a set to determine the string contains unique characters
+    >>> isUniqueSet("train")
+    True
+    >>> isUniqueSet("test")
+    False
+    """
+
+    hashSet = set()
+    for i in string:
+        if i in hashSet:
```
Comment (mateuszz0000) [inline strings/characters_unique.py:23]: `hashSet` -> `hash_set`

File: strings/characters_unique.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+def isUniqueSet(string: str) -> bool:
+    """
+    This function uses a set to determine the string contains unique characters
+    >>> isUniqueSet("train")
+    True
+    >>> isUniqueSet("test")
+    False
+    """
+
+    hashSet = set()
```
Comment (mateuszz0000) [inline strings/characters_unique.py:21]: We can use `set` in different way. We can just `return len(list) == len(set) <- pseudocode' to determine unique characters.

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**